File: src/com/iamunclexu/confs/TemplateConf.java
package com.iamunclexu.confs;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.io.IOException;

import freemarker.cache.FileTemplateLoader;
import freemarker.template.Configuration;

import static com.iamunclexu.confs.Constant.TEMPLATE_DOCUMENT;

public class TemplateConf {

    private static Logger LOGGER = LoggerFactory.getLogger(TemplateConf.class);
    private static TemplateConf templateConf;

    private static Configuration configuration;

    public static TemplateConf inst() {
        if (templateConf == null) {
            templateConf = new TemplateConf();
        }
        return templateConf;
    }

    public void init() {
        configuration = new Configuration();
        try {
            String templatePath = SysConf.FRONTEND_ROOT + TEMPLATE_DOCUMENT;

            LOGGER.info("Template dir is : " + templatePath);
            configuration.setTemplateLoader(new FileTemplateLoader(new File(templatePath)));
        } catch (IOException e) {
            LOGGER.error(e.getMessage());
        }
    }

    public static Configuration fetchConfiguration() {
        return configuration;
    }
}

File: src/com/iamunclexu/confs/TemplateConf.kt
package com.iamunclexu.confs

import com.iamunclexu.confs.Constant.TEMPLATE_DOCUMENT
import freemarker.cache.FileTemplateLoader
import freemarker.template.Configuration
import org.slf4j.LoggerFactory
import java.io.File
import java.io.IOException

class TemplateConf {

    fun init() {
        configuration = Configuration()
        try {
            val templatePath = "${SysConf.FRONTEND_ROOT}$TEMPLATE_DOCUMENT"

            LOGGER.info("Template dir is : $templatePath")
            configuration.templateLoader = FileTemplateLoader(File(templatePath))
        } catch (e: IOException) {
            LOGGER.error(e.message)
        }

    }

    companion object {
        lateinit var configuration: Configuration

        private val LOGGER = LoggerFactory.getLogger(TemplateConf::class.java)
        private var templateConf: TemplateConf? = null
            get() {
                if (field == null) {
                    field = TemplateConf()
                }
                return field
            }

        fun inst(): TemplateConf {
            return templateConf!!
        }

        fun fetchConfiguration(): Configuration? {
            return configuration
        }
    }
}
